.github/workflows/generate-changeset.yml

```diff
@@ -0,0 +1,38 @@
+name: Generate changeset
+on:
+  workflow_run:
+    workflows: ["trigger changeset generation"]
+    types:
+      - completed
+
+env:
+  CI: true
+  NODE_OPTIONS: "--max-old-space-size=4096"
+
+concurrency:
+  group: ${{ github.event.workflow_run.head_repository.full_name }}::${{ github.event.workflow_run.head_branch }}
+
+jobs:
+  version:
+    permissions: write-all
+    name: static checks
+    runs-on: ubuntu-22.04
+    if: (github.event.workflow_run.head_repository.full_name == 'gradio-app/gradio' && github.event.workflow_run.head_branch != 'main') || github.event.workflow_run.head_repository.full_name != 'gradio-app/gradio'
+    steps:
+      - id: 'get-pr'
+        uses: "gradio-app/github/actions/get-pr-branch@main"
```

Comment:
It's similar but not quite the same. I'm not 100% certain if that action would work for us as ours is tailored for workflow runs (and doesn't require any input), we use the branch name + repo owner to check the PR rather than a commit sha which is what i think that action is doing. It could work tho, not certain.